### PR TITLE
Don't drop user dir deltas when server leaves room

### PR DESCRIPTION
**Pull request by DMRobertson**
I spotted this when I was trying to do another refactor/tidy up of this code. Argh! Not sure if we ever hit this (I don't know the details of how `_handle_deltas` is hit---it presumably is sensitive to replication delays?).

Anyway, this squashes a bug and makes be feel better.

### CODE REVIEW

File: tests/unittest.py [hunk 1]
```diff
@@ -317,6 +317,15 @@ def wait_on_thread(self, deferred, timeout=10):
             self.reactor.advance(0.01)
             time.sleep(0.01)
 
+    def wait_for_background_updates(self) -> None:
+        """Block until we've written completed all background updates."""
```
Comment (clokep) [inline tests/unittest.py:321]: I think there's a word missing or something?

```suggestion
        """Block until all background updates have completed."""
```

**Review by clokep — APPROVED** [review]
Seems reasonable to me!

File: tests/unittest.py (hunk 1 above)
Comment (DMRobertson) [reply tests/unittest.py:321]: Ahh, thank you. I had in mind something like "block until we've written to the DB that all background updates have completed". But I think the former is an unnecessary detail.

Comment (clokep) [reply tests/unittest.py]: Yeah, I don't think the caller really cares about the details, just that they've all finished! (Also, thanks for abstracting this out!)

File: tests/storage/test_roommember.py [hunk 2]
```diff
@@ -197,9 +192,4 @@ def test_can_rerun_update(self):
         self.store.db_pool.updates._all_done = False
```
Comment (DMRobertson) [inline tests/storage/test_roommember.py:192]: I did have one concern here. Some of these explicitly reset the `_all_done` flag, and others don't. I haven't made the situation worse---apart from an extra stack frame the behaviour should be identical. But I wondered if it was was more consistent to reset the flag in the helper function.

File: tests/unittest.py [hunk 3]
```diff
@@ -317,6 +317,15 @@ def wait_on_thread(self, deferred, timeout=10):
             self.reactor.advance(0.01)
             time.sleep(0.01)
 
+    def wait_for_background_updates(self) -> None:
+        """Block until all background updates have completed."""
```
Comment (richvdh) [inline tests/unittest.py:321]:
```suggestion
        """Block until all background database updates have completed."""
```

File: tests/storage/test_roommember.py (hunk 2 above)
Comment (richvdh) [reply tests/storage/test_roommember.py:192]: :man_shrugging:

**Review by richvdh — APPROVED** [review]
lgtm

**Pull request merged**